plasmapy/classes/plasma.py

```diff
@@ -58,22 +58,152 @@ def __init__(self, domain_x, domain_y, domain_z):
         self.domain_shape = (len(self.x), len(self.y), len(self.z))
 
         # Initiate core plasma variables
-        self.density = np.zeros(self.domain_shape) * u.kg / u.m**3
-        self.momentum = np.zeros((3, *self.domain_shape)) * u.kg / (u.m**2 * u.s)
-        self.pressure = np.zeros(self.domain_shape) * u.Pa
-        self.magnetic_field = np.zeros((3, *self.domain_shape)) * u.T
+        self._density = np.zeros(self.domain_shape) * u.kg / u.m**3
+        self._momentum = np.zeros((3, *self.domain_shape)) * u.kg / (u.m**2 * u.s)
+        self._pressure = np.zeros(self.domain_shape) * u.Pa
+        self._magnetic_field = np.zeros((3, *self.domain_shape)) * u.T
 
+    """
+    Define getters and setters for variables.
+    """
+    ## ==== Core variables ====
+    # Density
+    @property
+    def density(self):
+        return self._density
+
+    @density.setter
+    @u.quantity_input
+    def density(self, density: u.kg/u.m**3):
+        """Sets the simulation's density profile to the specified array.
+        Other arrays which depend on the density values, such as the kinetic pressure,
+        are then redefined automatically.
+        
+        Parameters
+        ----------
+
+        density : ndarray
+            Array of density values. Shape and size must be equal to those of the simulation grid.
+            Must have units of density.
+
+        """
+
+        assert density.shape == self.grid_size, \
+            'Specified density array shape does not match simulation grid.'
+        self._density = density
```

Comment:
Doesn't the previous necessitate `self._pressure = self.pressure()` after this line here?